sample-app-compose/src/main/java/io/livekit/android/composesample/ParticipantItem.kt

```diff
@@ -1,80 +1,116 @@
 package io.livekit.android.composesample
 
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.runtime.*
+import androidx.compose.material.Icon
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.layout.onGloballyPositioned
-import androidx.compose.ui.viewinterop.AndroidView
-import com.github.ajalt.timberkt.Timber
-import io.livekit.android.renderer.TextureViewRenderer
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import io.livekit.android.composesample.ui.theme.BlueMain
+import io.livekit.android.composesample.ui.theme.NoVideoBackground
 import io.livekit.android.room.Room
-import io.livekit.android.room.participant.ParticipantListener
-import io.livekit.android.room.participant.RemoteParticipant
-import io.livekit.android.room.track.RemoteTrackPublication
-import io.livekit.android.room.track.RemoteVideoTrack
+import io.livekit.android.room.participant.Participant
 import io.livekit.android.room.track.Track
-import io.livekit.android.room.track.video.ComposeVisibility
+import io.livekit.android.room.track.VideoTrack
+import io.livekit.android.util.flow
 
 @Composable
 fun ParticipantItem(
     room: Room,
-    participant: RemoteParticipant,
+    participant: Participant,
+    modifier: Modifier = Modifier,
+    isSpeaking: Boolean,
 ) {
-    val videoSinkVisibility = remember(room, participant) { ComposeVisibility() }
-    var videoBound by remember(room, participant) { mutableStateOf(false) }
-    fun getVideoTrack(): RemoteVideoTrack? {
-        return participant
-            .videoTracks.values
-            .firstOrNull()?.track as? RemoteVideoTrack
-    }
 
-    fun setupVideoIfNeeded(videoTrack: RemoteVideoTrack, view: TextureViewRenderer) {
-        if (videoBound) {
-            return
-        }
+    val identity by participant::identity.flow.collectAsState()
```

Comment:
example of using the flows in compose

Reply:
this looks neat! how does it know when to refresh?

Reply:
So kotlin flows are basically an observable stream, and in this case it emits whenever the value is updated. That gets turned into a state variable with the `collectAsState()` method here.

On the compose side, there's just a huge amount of behind the scenes compiler side magic that goes on, but basically whenever a state variable is changed, it triggers a UI update.